src/clipboard.rs

```diff
@@ -14,24 +17,60 @@ use crate::window::UserEvent;
 
 type Result<T> = std::result::Result<T, Box<dyn Error + Send + Sync + 'static>>;
 
-static CLIPBOARD: OnceLock<Mutex<Box<dyn ClipboardProvider>>> = OnceLock::new();
+pub struct Clipboard {
+    clipboard: Box<dyn ClipboardProvider>,
+    #[cfg(target_os = "linux")]
+    selection: Box<dyn ClipboardProvider>,
+}
+
+static CLIPBOARD: OnceLock<Mutex<Clipboard>> = OnceLock::new();
 
 pub fn init(event_loop: &EventLoop<UserEvent>) {
     CLIPBOARD
         .set(Mutex::new(match event_loop.raw_display_handle() {
             #[cfg(target_os = "linux")]
             RawDisplayHandle::Wayland(WaylandDisplayHandle { display, .. }) => unsafe {
-                Box::new(wayland_clipboard::create_clipboards_from_external(display).1)
+                let clipboards = wayland_clipboard::create_clipboards_from_external(display);
```

Comment:
nit: could we destructure clipboards into more reasonable names rather than using clipboards.0 and clipboards.1 especially since they are passed in reverse order?